pyshtools/shio/icgem.py

```diff
@@ -0,0 +1,188 @@
+"""
+ICGEM-format read support
+"""
+
+from __future__ import absolute_import as _absolute_import
+from __future__ import division as _division
+
+import numpy as _np
+
+from pyshtools.utils.datetime import _yyyymmdd_to_year_fraction
+
+
+def _time_variable_part(epoch, ref_epoch, trnd, periodic):
+    """Return sum of the time-variable part of the coefficients
+
+    The formula is:
+    G(t) = G(t0) + trnd*(t-t0 ) +
+        asin1*sin(2pi/p1 * (t-t0)) + acos1*cos(2pi/p1 * (t-t0)) +
+        asin2*sin(2pi/p2 * (t-t0)) + acos2*cos(2pi/p2 * (t-t0))
+
+    This function computes all terms after G(t0).
+    """
+    delta_t = epoch - ref_epoch
+    trend = trnd * delta_t
+    periodic_sum = _np.zeros_like(trnd)
+    for period in periodic:
+        for trifunc in periodic[period]:
+            coeffs = periodic[period][trifunc]
+            if trifunc == 'acos':
+                periodic_sum += coeffs * _np.cos(2 * _np.pi / period * delta_t)
+            elif trifunc == 'asin':
+                periodic_sum += coeffs * _np.sin(2 * _np.pi / period * delta_t)
+    return trend + periodic_sum
+
+
+def read_icgem_gfc(filename, errors=None, lmax=None, epoch=None):
+    """Read spherical harmonic coefficients from an ICGEM GFC ascii-formatted file.
+
+    This function only reads files with the gravity field spherical
```

Comment:
It does, the ICGEM-format can also be used for ocean/atmospheric tides, but this function will not be able to read it.